File: flutter_sdk/lib/src/data/services/camera/camera_service.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

import 'package:chat_flutter_sdk/src/common/result.dart';
import 'package:flutter/services.dart';
import 'package:image_picker/image_picker.dart';

abstract class CameraService {
  Future<Result<XFile?>> pickImage(ImageSource source);
  Future<Result<Unit>> saveImage(String path);
  Future<Result<Unit>> deleteImage(String path);
}

File: typescript-sdk/packages/core/src/data/services/camera/camera-service.ts
// Copyright (c) Yalochat, Inc. All rights reserved.

import type { Result } from '../../../common/result.js';
import type { ImageData } from '../../../domain/image-data.js';

export interface CameraService {
  /**
   * Opens a file picker or camera capture and returns image data.
   * Returns undefined if the user cancelled.
   */
  pickImage(): Promise<Result<ImageData | undefined>>;

  /** Saves an image at the given path and returns its URL. */
  save(fileName: string, data: Uint8Array, mimeType: string): Promise<Result<string>>;

  /** Deletes a previously saved image by fileName. */
  delete(fileName: string): Promise<Result<void>>;
}
